frontend/src/hooks/useCustomEvent.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+
+/**
+ * A hook that provides methods to emit and listen to custom DOM events.
+ * @param eventName The name of the event to listen for and dispatch.
+ */
+export const useCustomEvent = (
+    eventName: string,
```

Comment:
I think leaving it as string for now is fine, but we could consider using an enum later if we find that we use this a lot, to avoid propagating something that we're not catching. Not that that would guarantee anything to be caught, but it might catch some minor typos or case inconsistencies (are dom events case sensitive?).

Reply:
Normal usage pattern for now is to use this new hook only once in the useLastViewedProject so whatever we put there (even with typo) is consistently used by all useLastViewedProject() invocations.

Reply:
events are case sensitive: https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#type

Reply:
Ah, good point! And because you're automatically listening for the event you created, it doesn't really matter. Yeah, I get it now 😄